snowshu/core/docker.py

```diff
@@ -144,20 +116,80 @@ def get_stopped_container(  # noqa pylint: disable=too-many-arguments
-                            image.tag(f'{target_adapter.DOCKER_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
+                                target_adapter.BASE_DB_IMAGE, platform=f'linux/{arch}')
+                            image.tag(f'{target_adapter.BASE_DB_IMAGE.split(":")[0]}:{arch}')  # noqa pylint: disable=use-maxsplit-arg
+
+                except ConnectionError as error:
+                    logger.error(
+                        'Looks like docker is not started, please start docker daemon\nError: %s', error)
+                    raise
+
+                tagged_container_name = f'{name}_{arch}'
+                logger.info(
+                    f"Creating stopped container {tagged_container_name}...")
+                self.remove_container(tagged_container_name)
+
+                container = self.client.containers.create(
+                    image.tags[0],
+                    target_adapter.DOCKER_START_COMMAND,
+                    network=network.name,
+                    name=tagged_container_name,
+                    hostname=target_adapter._credentials.host,  # noqa pylint: disable=protected-access
+                    ports=port_dict,
+                    environment=envars,
+                    labels=dict(
+                        snowshu_replica='true',
+                        target_adapter=target_adapter.CLASSNAME,
+                        source_adapter=source_adapter),
+                    detach=True,
+                    volumes={replica_volume.name: {
+                        'bind': f'{DOCKER_REPLICA_MOUNT_FOLDER}',
+                        # Make sure passive container does not mess up common volume
+                        'mode': 'rw' if arch == arch_list[0] else 'ro'
+                    }},
+                    working_dir=DOCKER_WORKING_DIR
+                )
+                logger.info(f"Created stopped container {container.name}.")
+
+                logger.info(
+                    f'Connecting {container.name} to bridge network..')
+                self._connect_to_bridge_network(container)
+                logger.info(
+                    f'Connected. Starting created container {container.name}...')
 
-                else:
+                try:
+                    container.start()
+                except docker.errors.APIError as error:
+                    if 'port is already allocated' in error.explanation:
+                        logger.exception('One of the ports used by snowshu_target is '
+                                         'already allocated, stop extra containers and rerun')
+                    raise
+
+                logger.info(
+                    f'Container {container.name} started, running initial setup...')
+                self._run_container_setup(container, target_adapter)
+                logger.info(f'Container {container.name} fully initialized.')
+
+                if len(arch_list) > 1:
+                    container.stop()
```

Comment:
Would it make sense to move this block of code back into something like the `get_stopped_container` function? It ends up being identical for both the `if` and the `else` blocks here.

Reply:
Moved into other method, together with setup.